networks/edsl_examples/reorgyolo.py

```diff
@@ -0,0 +1,189 @@
+    arrayLen = len(arrayIn)
+    arrayOut = np.zeros(arrayLen)
+    print("C is " + str(C) + "stride is " + str(stride))
+    out_c = C // (stride * stride)  #out_c = 1
+    for b in range(batch):
+        for k in range(C):
+            for j in range(H):
+                for i in range(W):
+                    in_index = i + W * (j + H * (k + C * b))
+                    c2 = k % out_c
+                    offset = k // out_c
+                    w2 = i * stride + offset % stride
+                    h2 = j * stride + offset // stride
+                    out_index = int(w2 + W * stride * (h2 + H * stride * (c2 + out_c * b)))
+                    if forward:
+                        arrayOut[out_index] = arrayIn[in_index]
+                    else:
+                        arrayOut[in_index] = arrayIn[out_index]
+    return arrayOut
+
+
+def reorgyolo_comparison_nodivmod(arrayIn, batch, C, H, W, stride, forward=False):
+    arrayLen = len(arrayIn)
+    arrayOut = np.zeros(arrayLen)
+    print("C is " + str(C) + "stride is " + str(stride))
+    out_c = C // (stride * stride)
+    _c1_quotient_range = int(C // (out_c))
+    for n1 in range(batch):
+        for w1 in range(W):
+            for h1 in range(H):
+                for c2 in range(out_c):
+                    for _w2_quotient in range(_c1_quotient_range // stride):
+                        for _w2 in range(stride):
+                            _c1 = _w2 + _w2_quotient * stride
+                            c1 = c2 + (_c1 * out_c)
+                            in_index = w1 + W * (h1 + H * (c1 + C * n1))
+                            w2 = w1 * stride + _w2
+                            h2 = h1 * stride + _w2_quotient
+                            out_index = int(w2 + W * stride * (h2 + H * stride *
+                                                               (c2 + out_c * n1)))
+                            if forward:
+                                arrayOut[out_index] = arrayIn[in_index]
+                            else:
+                                arrayOut[in_index] = arrayIn[out_index]
+    return arrayOut
+
+
+def reorgyolo(I, stride, forward):
+    dims = I.shape.int_dims
+    N = dims[0]
+    C = dims[1]
+    H = dims[2]
+    W = dims[3]
+
+    out_c = C // (stride * stride)
+
+    N_out = N
+    C_decrease = int(C // (stride * stride))
+    C_increase = int(C * (stride * stride))
+    _c1_quotient_range = int(C // (C_decrease))
+    _w2_quotient_range = int(_c1_quotient_range // stride)
+    N_in, C_in, H_in, W_in = edsl.TensorDims(4)
+    n1, w1, h1, c2, _w2_quotient, _w2 = edsl.TensorIndexes(6)
+    I.bind_dims(N_in, C_in, H_in, W_in)
+
+    #if forward:
+    O = edsl.TensorOutput(N_out, C_decrease, int(H * stride), int(W * stride))
+    O[n1, c2, h1 * stride + _w2_quotient, w1 * stride +
+      _w2] = I[n1, c2 + ((_w2 + _w2_quotient * stride) * out_c), h1, w1]
+    # else:
+    #     O = edsl.TensorOutput(N_out, C_increase, int(H/stride), int(W/stride))
```

Comment:
This `H//stride` and `W//stride` code might be a place to look at for `forward == false` problems

Reply:
To clarify: I mean this in cases where the spatial dims aren't evenly divided by the stride. If stride divides H and W evenly I would not expect these to cause problems.